.github/workflows/bench.yml

```diff
@@ -82,7 +82,38 @@ jobs:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           auto-push: true
           # Ratio indicating how worse the current benchmark result is.
-          # 150% means if last build took 40s and current takes 50s, it will trigger an alert
+          # 150% means if last build took 40s and current takes >60s, it will trigger an alert
+          alert-threshold: "150%"
+          fail-on-alert: true
+          # Enable alert commit comment
+          comment-on-alert: true
+          # Mention @jchavarri in the commit comment
+          alert-comment-cc-users: '@jchavarri'
+
+      - name: Create synthetic benchmark
+        working-directory: bench
+        run: opam exec -- ../_boot/dune.exe exec ./gen_synthetic.exe -- -n 6 synthetic
+
+      - name: Run synthetic benchmark
+        working-directory: bench/synthetic
+        run: ../gen-benchmark.sh 'opam exec -- ../../_boot/dune.exe build ./main.exe' 'opam exec -- ../../_boot/dune.exe clean' 'synthetic build time (${{ runner.os }})' > synthetic-benchmark-result.json
```

Comment:
Otherwise you don't get the "full picture" of the build. But we could go the other way: prepare the build first by running once the build, and only measure how long it takes to build again. Would that be better?

Reply:
I'm not sure which one is better. It's fine as is for now.